### PR TITLE
Resolve return results to "Relay"

### LINKED ISSUE #73: Resolve results return
### Is this the right issue type?

- [x] Yes, I'm planning work for this project team.

### Summary

Now Resolve has made it's query, it needs to return and POST it back to Relay.

### Acceptance Criteria

- [x] Send Result payload to Relay web API.
- [x] Results payload matches RQuest Result schema.

### Tasks

- [x] Post results to `task/result/{uuid}/{collectionId}`
- [x] Handle timeout errors (within the 5 minute timeout)

### Confirm creation

- [x] This issue is ready

### PR DESCRIPTION
**Pull request by AndrewThien**
## Pull Request Contents

| |
|-|
♻️ Refactor
✨ Feature

## Description

This PR did the refraction to make the resolve.cli and resolve.daemon use the same logic of receiving the input, resolve the query but one will save input as JSON, and another will send the JSON to the designated URL (assumingly: https://relay/task/result/{result.uuid}/{result.collection_id}) (tested successfully with a fake server, return 200 and the body as the json data sent) . Resolve will handle the error after "a few times" of trying to send back the results to Relay.

This PR also added tests using Pytest to compare the results of Resolve.

The resolve daemon service in Docker compose now has the env. vars to interact with DB and other services.

## Related Issues or other material

- [x] This PR relates to one or more issues, detailed below

- Related Issue #73
- Closes #73

## ✅ Added/updated tests?

- [x] This PR contains relevant tests

### CODE REVIEW

File: app/resolve/pyproject.toml [hunk 1]
```diff
@@ -29,6 +29,7 @@ trino={version = "^0.328.0", extras = ["sqlalchemy"]}
 python-dotenv = "1.0.1"
 mysqlclient="^2.1.1"
 pyodbc="^4.0.35"
+pytest="8.3.3"
```
Comment (AndyRae) [inline app/resolve/pyproject.toml:32]: Please add this to a `test` group - 
https://python-poetry.org/docs/managing-dependencies/#dependency-groups

File: app/resolve/test/test_return.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+from core.rquest_dto.group import Group
+from core.rquest_dto.rule import Rule
+from dotenv import load_dotenv
+import os
+import core.config as config
+
+load_dotenv()
+
+
+@pytest.fixture
```
Comment (AndyRae) [inline app/resolve/test/test_return.py:16]: Tests! Wonderful.
Let's do an issue to run these in CI, and make them easier to run locally too (for example vscode integration is nice)

File: app/resolve/daemon/__init__.py [hunk 3]
```diff
@@ -1,2 +1,26 @@
...
+from core.rquest_dto.result import RquestResult
+
+
 def main() -> None:
-    print("hello rabbit world!")
+    logger = logging.getLogger(config.LOGGER_NAME)
+    result = execute_query(parser)
+    if not isinstance(result, RquestResult):
+        raise TypeError("Payload does not match RQuest result schema")
+    destination = environ.get("RESOLVE_DAEMON_POST_URL")
```
Comment (AndyRae) [inline app/resolve/daemon/__init__.py:17]: I believe a `settings.py` is in #72 - but worth adding it here as this is ready. 
So a new `settings.py` in `core`, get the environment variable in there, and import that in this file.

Let's name this environment variable better too, will follow up with a comment.

**Review by AndyRae — CHANGES_REQUESTED** [review]

File: app/resolve/daemon/__init__.py (hunk 3 above)
Comment (AndyRae) [reply app/resolve/daemon/__init__.py:17]: `TASK_API_BASE_URL` would work - and then this code can append the path.

File: app/resolve/daemon/__init__.py [hunk 4]
```diff
@@ -1,2 +1,26 @@
...
+    logger = logging.getLogger(config.LOGGER_NAME)
+    result = execute_query(parser)
+    if not isinstance(result, RquestResult):
+        raise TypeError("Payload does not match RQuest result schema")
+
+    return_url = f"{TASK_API_BASE_URL}/{result.uuid}/{result.collection_id}"
+
+    for i in range(4):
+        response = requests.post(return_url, data=result.to_dict())
+        if response.status_code == 200:
```
Comment (beforan) [inline app/resolve/daemon/__init__.py:22]: On reflection, I think `200` is too specific here.

At minimum `409 Conflict` should not cause further retries. (as it means results have already been received).

For now let's say 2xx and 409 don't retry? (I'm open to not retrying for any 4xx)

**Review by beforan — CHANGES_REQUESTED** [review]

File: app/resolve/daemon/__init__.py (hunk 4 above)
Comment (AndrewThien) [reply app/resolve/daemon/__init__.py]: thanks! pushed the changes

**Review by beforan — APPROVED** [review]

**Review by AndyRae — APPROVED** [review]

**Pull request merged**